src/types/wrappers/MaxMinDrivingForceModel.jl

```diff
@@ -0,0 +1,307 @@
+"""
+$(TYPEDEF)
+
+Return a [`MaxMinDrivingForceModel`](@ref) that can be used to perform max-min
+driving force analysis. It is based on the work by Noor, et al., "Pathway
+thermodynamics highlights kinetic obstacles in central metabolism.", PLoS
+computational biology, 2014.
+
+When [`flux_balance_analysis`](@ref) is called in this type of model, the
+max-min driving force algorithm is solved. It returns the Gibbs free energy of
+the reactions, the concentrations of metabolites, and the actual maximum minimum
+driving force across all reactions in the model. The optimization problem solved
+is:
+```
+max min -ΔᵣG
+s.t. ΔᵣG = ΔrG⁰ + R T S' ln(C)
+     ΔᵣG ≤ 0
+     ln(Cₗ) ≤ ln(C) ≤ ln(Cᵤ)
+```
+where `ΔrG` are the Gibbs energies dissipated by the reactions, R is the gas
+constant, T is the temperature, S is the stoichiometry of the model, and C is
+the vector of metabolite concentrations (and their respective lower and upper
+bounds).
+
+In case no feasible solution exists, `nothing` is returned.
+
+Since biochemical thermodynamics are assumed, the `proton_ids` and `water_ids`
+*need* to be specified so that they can be ignored in the calculations.
+Effectively this assumes an aqueous environment at constant pH is used.
+
+# Fields
+$(TYPEDFIELDS)
+"""
+Base.@kwdef mutable struct MaxMinDrivingForceModel <: AbstractModelWrapper
+    "A dictionary mapping ΔrG⁰ to reactions."
+    reaction_standard_gibbs_free_energies::Dict{String,Float64} = Dict{String,Float64}()
+
+    "A cycle-free reference flux solution that is used to set the directions of the reactions."
+    flux_solution::Dict{String,Float64} = Dict{String,Float64}()
```

Comment:
this should document how to get the reference solution. Is it variable values or reaction flux values or something else? Can this work on more complicated reactions? (If not we're all ok, just make a note that "this can't yet work on multivariable/split reactions")